### DISCUSSION TITLE
Just got TS100 (Upgraded) -- Need Advice

### DISCUSSION THREAD

**Discussion by PhoenixViridian** [Ralim/IronOS#1319] [General]
I just picked up a TS100 Upgraded from Amazon... Specifically this one: https://www.amazon.com/dp/B01MD12DYT

I have heard that the IronOS firmware is dangerous and can blow out the OLED screen, is this true? Or is this actually the best firmware for a TS100? 

As well, Is it as simple as following the documentation to upgrade the firmware?

**Comment by PhoenixViridian** [comment]
Anyone know if the above is still true?

**Comment by Ralim** [comment]
Hia,

(1)

There is no such thing as "upgraded" TS100's. That's purely marketing.

(2)

No real damaged is known (if a firmware release has risks, it's download link is pulled).


IronOS can't do anything to your screen that the stock firmware _cant_. The cheap OLED screens have a shelf life before they die due to manufacturing defects. Running your iron or not doesn't change that at all.

Additionally they may burn in if you run the screen at a high brightness (you can adjust this in firmware, the default is sensible).

Most OLED issues have not been the fault of the firmware.

Except; this firmware runs the i2c communications bus with the screen faster than the stock firmware (for nice UI effects). If your iron has poor assembly on the screen connector you may run into glitching effects on the OLED. The fix for this is to fix the poor factory soldering job.
It's not really the firmware, but more a failed QC at the factory.


In general updating is as simple as the docs. However miniware has changed to stm32 clones recently and their bootloader for upgrading is rather temperamental.

Check what DFU version is on your iron; if it's 3.47 or newer you have an iron with the clone MCU. It just means upgrading takes a few tries; otherwise it will work fine. (Look at issues tab for some reports of this).

Sadly as the DFU bootloader isn't in our control; I can't really fix this.

**Comment by PhoenixViridian** [comment]
Welp, I had bootloader 3.49 and it loaded instantly. It seems this one might not be a clone, it shows up as a STM32.

**Reply by Ralim** [reply]
They all show up as "stm32" devices; but so long as it loaded you should be fine.

**Reply by River-Mochi** [reply]
> Welp, I had bootloader 3.49 and it loaded instantly. It seems this one might not be a clone, it shows up as a STM32.

I have seen a bunch of  people had to reheat/reflow their OLED screens because the solder job at factory was poor and left cold joints. reflowing fixed issues. this is why there are so many Youtube videos showing people how to reflow or even replace the OLed screens on their  TS100 or other similar irons.   the replacement screens are about  $3-$5. 
IronOS  is a better firmware from stock but it will exposed  poor manufacturing defects like this due to what Ralim said, so in a way it's good to run it and see right away if you have issue with the Iron for  return warranty even if you have to put original OS back on.  
It's fairly easy to reflow  the screen or resolder it if you follow the videos (look for the  replacement videos ).  many times reflow solder is enough to get all the legs well soldered. But dimming screen does help prolong life. I ordered extra screens for my Pinecil just to have on hand since it takes 2 months to slow ship them.

Also as a reminder,  this firmware is free  and Ralim is volunteer also like all of us. this is not  official support for TS100,  so it may take several days for people to read  your request or get back to you. it's 100% community driven so people need to be patient if someone has time to answer.

There is also the official seller of TS100 that you bought it from that you coud contact  since they are responsible for your TS100. 

**Where are people saying that Ralim's IronOS  blows out the screen?**  I  have not read this yet so I'm curious.  If they knew how/why this was happening (cold solder joints/ poor oled lifespans) and separately - the  faster screen that IRonOS uses to make things look nicer that expose poor factory soldering, perhaps people would not be saying don't use IronOS it blows out your screen.  that seems like mis-information they are spreading or not listing all the details thoroughly from lack of understanding.

**Comment by Gadgetoid** [comment]
Have a bootloader 3.49 Iron with GD32F103 here. Once I got the steps right (painful on Linux) the firmware installed fine first try. (to be fair the docs do provide a script but I only spotted that *after* I flashed). No issues with my OLED.
